src/forms/incident-create.ts

```diff
@@ -0,0 +1,162 @@
+            },
+        },
+    } as AppForm;
+}
+
+export async function addIncidentFromCommand(call: AppCallRequest) {
+    const values = call.values as CreateIncidentFormCommandType;
+    const i18nObj = configureI18n(call.context);
+
+    if (!values?.incident_impacted_service && !values?.incident_title) {
+        throw new Error(i18nObj.__('forms.incident-create.add-incident-exception'));
+    }
+
+    const oauth2: Oauth2App = call.context.oauth2 as Oauth2App;
+    const tokenOpts: PagerDutyOpts = { token: <string>oauth2.user?.token, tokenType: 'bearer' };
+
+    const serviceOpts: AppSelectOption[] = await getServiceOptionList(tokenOpts, call.context);
+    const assignToOpts: AppSelectOption[] = await getUsersOptionList(tokenOpts, call.context);
+
+    const service = serviceOpts.find((b) => b.value === values.incident_impacted_service || b.label === values.incident_impacted_service);
+    const assign = assignToOpts.find((b) => b.value === values.incident_assign_to || b.label === values.incident_assign_to);
+
+    const valuesData: CreateIncidentFormModalType = {
+        incident_impacted_service: <AppSelectOption>service,
+        incident_title: values.incident_title,
+        incident_description: values.incident_description,
+        incident_assign_to: <AppSelectOption>assign,
+    };
+
+    await postIncidentMethod(valuesData, tokenOpts, call.context);
+}
+
+export async function submitCreateIncident(call: AppCallRequest): Promise<any> {
+    const values = call.values as CreateIncidentFormModalType;
+    const i18nObj = configureI18n(call.context);
+
+    const oauth2: Oauth2App = call.context.oauth2 as Oauth2App;
+    const tokenOpts: PagerDutyOpts = { token: <string>oauth2.user?.token, tokenType: 'bearer' };
+
+    if (!values?.incident_impacted_service && !values?.incident_title) {
+        throw new Error(i18nObj.__('forms.incident-create.create-incident-exception'));
+    }
+
+    await postIncidentMethod(values, tokenOpts, call.context);
+}
+
+async function postIncidentMethod(values: CreateIncidentFormModalType, pdOpt: PagerDutyOpts, context: AppContext) {
+    const pdClient: PartialCall = api({ token: pdOpt.token, tokenType: pdOpt.tokenType });
+    const i18nObj = configureI18n(context);
+
+    const incident: PostIncident = {
+        incident: {
+            type: 'incident',
+            title: values.incident_title,
+            service: {
+                id: values.incident_impacted_service.value,
+                type: 'service_reference',
+            },
+        },
+    };
+
+    if (Boolean(values.incident_description)) {
+        incident.incident.body = {
+            type: 'incident_body',
+            details: values.incident_description,
+        };
+    }
+
+    if (Boolean(values.incident_description)) {
+        incident.incident.body = {
+            type: 'incident_body',
+            details: values.incident_description,
+        };
+    }
+    if (Boolean(values.incident_assign_to)) {
```

Comment:
Optional

These calls to `Boolean` redundant

```suggestion
    if (values.incident_assign_to) {
```